scripts/prepare_network.py

```diff
@@ -282,13 +325,28 @@ def set_line_nom_max(n, s_nom_max_set=np.inf, p_nom_max_set=np.inf):
         if "Co2L" in o:
             m = re.findall("[0-9]*\.?[0-9]+$", o)
             if len(m) > 0:
-                co2limit = float(m[0]) * snakemake.config["electricity"]["co2base"]
-                add_co2limit(n, co2limit, Nyears)
-                logger.info("Setting CO2 limit according to wildcard value.")
+                if snakemake.config["electricity"]["automatic_emission"]:
+                    emission_year = snakemake.config["electricity"]["automatic_emission_base_year"]
+                    filename = download_emission_data()
+                    co2limit = emission_extractor(filename, emission_year)
+                    co2limit = co2limit * float(m[0])
+                    add_co2limit(n, co2limit, Nyears)
+                    logger.info("Setting CO2 limit according to 1990 base year.")
+                else:
+                    co2limit = float(m[0]) * snakemake.config["electricity"]["co2base"]
+                    add_co2limit(n, co2limit, Nyears)
+                    logger.info("Setting CO2 limit according to wildcard value.")
             else:
-                co2limit = snakemake.config["electricity"]["co2limit"]
-                add_co2limit(n, co2limit, Nyears)
-                logger.info("Setting CO2 limit according to config value.")
+                if snakemake.config["electricity"]["automatic_emission"]:
+                    emission_year = snakemake.config["electricity"]["automatic_emission_base_year"]
+                    filename = download_emission_data()
+                    co2limit = emission_extractor(filename, emission_year)
+                    add_co2limit(n, co2limit, Nyears)
+                    logger.info("Setting CO2 limit according to 1990 base year.")
+                else:
+                    co2limit = snakemake.config["electricity"]["co2limit"]
+                    add_co2limit(n, co2limit, Nyears)
+                    logger.info("Setting CO2 limit according to config value.")
```

Comment:
This block seems somehow duplicated.
We may revise this block, including the after the `if len(m) > 0` as follows
```python
m = re.findall("[0-9]*\.?[0-9]+$", o)
if snakemake.config["electricity"]["automatic_emission"]:
    co2limit = ... [your calculations]
    logger.info(f"Setting the .... {target year}....")
elif len(m) > 0:
   co2limit = float(m[0]) * snakemake.config["electricity"]["co2base"]
   logger.info("Setting CO2 limit according to wildcard value.")
else:
   co2limit = snakemake.config["electricity"]["co2limit"]
   logger.info("Setting CO2 limit according to config value.")
add_co2limit(n, co2limit, Nyears)
```